include/envoy/http/filter.h

```diff
@@ -426,11 +426,15 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    *                       response headers.
    * @param grpc_status the gRPC status code to override the httpToGrpcStatus mapping with.
    * @param details a string detailing why this local reply was sent.
+   * @param retain_http_status_for_grpc bool, which if true indicates that the called service's
+   *                      HTTP response code will be passed along to gRPC clients.
+   *                      No effect for HTTP clients.
    */
   virtual void sendLocalReply(Code response_code, absl::string_view body_text,
                               std::function<void(ResponseHeaderMap& headers)> modify_headers,
                               const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
-                              absl::string_view details) PURE;
+                              absl::string_view details,
+                              bool retain_http_status_for_grpc = false) PURE;
```

Comment:
Why isn't this an options struct as elsehwere? A little uncomfortable proliferating more args here.

Reply:
Friendly ping